core/shared/src/main/scala/fs2/concurrent/Topic.scala

```diff
@@ -208,7 +208,7 @@ object Topic {
         }
 
         def publish: Pipe[F, A, Nothing] = { in =>
-          (in ++ Stream.exec(close.void))
+          in.onFinalize(close.void)
```

Comment:
Do we need the same fix in [Channel](https://github.com/typelevel/fs2/blob/abe9b03e205d05d41ca2950e0d9f5db7c77a26cf/core/shared/src/main/scala/fs2/concurrent/Channel.scala#L154)

Reply:
I made the change, it should also fix #3485. But as @ValdemarGr [pointed out](https://github.com/typelevel/fs2/issues/3485#issuecomment-2388220538), the exit case might not be correct.

Reply:
Here is an attempt to handle cancelation and error in Channel: e1d7b0925. What do you think?

Reply:
Looks good to me!

Reply:
I added this commit. Regarding Topic, I don't know how to implement the same behavior without exposing `raiseError` and `canceled` to Channel.

Reply:
I ended up adding `raiseError` and `cancel` to `Channel`, but now I'm facing mima issues. Is it acceptable to add new methods?